### PR TITLE
Update the Deployment section:

### PR DESCRIPTION
**Pull request by borys-belinsky**
1. Removing duplicated information in different files.
2. Optimizing cross-references: ensuring that the Deployment section contains all essential information
3. Extending the glossary with PMM related terminology.

### CODE REVIEW

File: doc/source/architecture.rst [hunk 1]
```diff
@@ -157,31 +156,8 @@ For more information, see :ref:`run-server`.
...
-
-Typical Scenario
-****************
-
-It is more typical to have several MySQL and MongoDB server instances
-distributed over different hosts.
-In this case, you can run *PMM Server* on a dedicated monitoring host,
-and install *PMM Client* on every database host that you want to monitor.
-Data from hosts will be aggregated on the PMM Server.
+.. DEPRECATED: moving deployment related information to the dedicated section
```
Comment (michaelcoburn) [inline doc/source/architecture.rst]: "to a dedicated section" or "to the dedicated deployment section"

File: doc/source/deploy/index.rst [hunk 2]
```diff
@@ -1,55 +1,277 @@
...
+|company-name| provides |product-abbrev| client packages through
+software repositories of popular Linux distributions:
 
-   * :ref:`Run PMM Server as a virtual appliance <run-server-ova>`
+* :ref:`DEB packages for Debian or Ubuntu <install-client-apt>`
+* :ref:`RPM packages for Red Hat or CentOS <install-client-yum>`
 
-   * :ref:`Run PMM Server using Amazon Machine Image (AMI) <run-server-ami>`
+It is recommended that you install your |product-abbrev| client by using the
+software repository of your system. If this options does not work for you,
```
Comment (michaelcoburn) [inline doc/source/deploy/index.rst]: "If this option does not" - drop the s

File: doc/source/architecture.rst (hunk 1 above)
Comment (borys-belinsky) [reply doc/source/architecture.rst]: "To the dedicated deployment section". It's a comment anyway which will not be rendered

File: doc/source/deploy/index.rst [hunk 3]
```diff
@@ -1,55 +1,277 @@
...
+   $ sudo apt-get update && sudo apt-get install pmm-client
+
+* For RedHat or CentOS::
+
+   $ yum update pmm-client
+
+If you have installed your |product-abbrev| client manually, you need to remove
+it and then download and install a newer version:
 
 Removing
```
Comment (michaelcoburn) [inline doc/source/deploy/index.rst]: We say earlier "Installing clients" - I think we should say "Removing clients" - Removing by itself seems to just  be begging - remove what?

File: doc/source/deploy/index.rst [hunk 4]
```diff
@@ -1,55 +1,277 @@
...
+     $ sudo yum remove pmm-client
+  
+- Removing a package installed by using APT:
+
+  .. code-block:: bash
 
-For more information about removing PMM, see the following:
+     $ sudo apt-get remove pmm-client
+  
+- Removing a directly installed RPM package:
```
Comment (michaelcoburn) [inline doc/source/deploy/index.rst]: drop word "directly" - not sure what value that adds to this sentence? how else did you add it?

File: doc/source/deploy/index.rst [hunk 5]
```diff
@@ -1,55 +1,277 @@
...
 
-* :ref:`remove-server`
-* :ref:`remove-client`
+  .. code-block:: bash
+
+     $ rpm -e pmm-client
+
+- Removing a directly installed DEB package:
```
Comment (michaelcoburn) [inline doc/source/deploy/index.rst]: drop "directly" - see comment on RPM above

File: doc/source/deploy/index.rst [hunk 6]
```diff
@@ -1,55 +1,277 @@
...
+<https://www.percona.com/downloads/pmm-client>`_ page.
 
-#. :ref:`Install PMM Client <install-client>`
-   on every MySQL and MongoDB instance
-   that you want to monitor.
+In addition to DEB and RPM packages, this site also offers:
 
-   Percona provides *PMM Client* packages for automatic installation
-   from software repositories on the most popular Linux distributions:
+* Generic tarballs that you can extract  and run the included ``install`` script.
```
Comment (michaelcoburn) [inline doc/source/deploy/index.rst]: 1. remove space between "extract  and"
2. "and run **using** the included"

File: doc/source/deploy/index.rst [hunk 7]
```diff
@@ -1,55 +1,277 @@
...
+separately. First, remove all monitored services by using the
+:program:`pmm-admin remove` command (see :ref:`pmm-admin-rm`). Then you can
+remove each |product-abbrev| client and the |product-abbrev| server.
+
+.. rubric:: Removing the |product-abbrev| Client
+
+The exact procedure of removing the |product-abbrev| client, depends
+on the method of installation:
+
+- Removing a package installed by using YUM:
```
Comment (michaelcoburn) [inline doc/source/deploy/index.rst]: "Remove an installed package using YUM" ?

File: doc/source/deploy/index.rst [hunk 8]
```diff
@@ -1,55 +1,277 @@
...
+
+  .. code-block:: bash
+
+     $ sudo yum remove pmm-client
+  
+- Removing a package installed by using APT:
```
Comment (michaelcoburn) [inline doc/source/deploy/index.rst]: "Remove an installed package using APT"

File: doc/source/deploy/index.rst (hunk 4 above)
Comment (borys-belinsky) [reply doc/source/deploy/index.rst]: rpm packages are normally managed by using the package manager such as yum. In this case (same as DEB and APT)  the user has installed the package directly without using the package manager. That's why "directly". Aptitude uses the term "manually installed" if a deb package was installed by using dpkg -i.

File: doc/source/deploy/index.rst (hunk 5 above)
Comment (borys-belinsky) [reply doc/source/deploy/index.rst]: dropping "directly" and using "manually" which is more appropriate.

**Pull request merged**